msamp/nn/distributed.py

```diff
@@ -133,6 +133,11 @@ def _reduce_bucket(self, bucket_id):
             param_ids = self.bucket_to_param_ids[bucket_id]
             params = [self.parameters[i] for i in param_ids]
             grads = [p.grad for p in params]
+            wgrad_qtype = Dtypes.kfloat8_e4m3
```

Comment:
When using TE, it will not set meta for gradient. So we just create a new meta for gradient here?